### ISSUE TITLE
[feature request] support for gitlab wiki pages

### ISSUE THREAD

**Issue by pReya** [gis-ops/wordpress-markdown-git#23]
Aside from regular markdown files, Gitlab also supports wiki pages, which behave a little differently from regular repo files. There's a separate API endpoint for wiki pages: https://docs.gitlab.com/ee/api/wikis.html

Would be great if these could be embedded as well, since these often contain static markdown content which is a good candidate to embed in a website.

**Comment by nilsnolde** [comment]
Sounds helpful. I won’t work in that direction myself (no use for us), but I’d be happy to review a PR if you wanna give it a shot.

**Comment by pReya** [comment]
@nilsnolde I might give it a try. Would you prefer to use a separate short code (like git-gitlabwiki) or to detect this from the URL structure and use the same shortcode?

**Comment by pReya** [comment]
@nilsnolde Sorry, this is slightly off topic, but I'm very interested in your reasoning: When you started this plugin, why did you chose to get the Git content via the API and not the regular raw web view that Gitlab/Github offer? (e.g. https://raw.githubusercontent.com/gis-ops/wordpress-markdown-git/master/README.md)

Wouldn't it be easier to use the web view, because you don't need to parse and split the URLs and just request the given URL directly?

Would you be interested in adding a "direct/raw" shortcode or parameter, which does exactly that? Just requests the given URL and sends it to the GitHub API to be parsed? This would also allow to embed Markdown files outside of git repos.

**Comment by nilsnolde** [comment]
You're raising good points. 

> Wouldn't it be easier to use the web view, because you don't need to parse and split the URLs and just request the given URL directly?

Not sure ;) Indeed that would eliminate a lot of the complexity, esp from provider specific implementations! Now it's a little late to change without breaking anything, but I'd be happy to have the direct/raw view and maybe eventually change it all to that logic. Be aware of private repos though, all providers differ here again in auth specifics. But surely easier than the current URL parsing/splitting!

Long story short: definitely would welcome that. Probably (for now) also best to have separate shortcodes for that (`git-<provider>-markdownraw`?). What do you think?

> Would you prefer to use a separate short code/provider (like git-gitlabwiki) 

Yep, I like that better and easier to implement I guess. Feel free to skip the other providers for now (goes for both `raw` and `wiki`), as long as you can implement it fairly generic (as much in base classes as possible).

**Comment by nilsnolde** [comment]
btw, feel free to change code as well if you feel there would be more elegant ways to do stuff. not really a PHP/WP guy here, this repo powers our tutorials section, that's the only reason it exists;)

**Comment by pReya** [comment]
@nilsnolde Thanks for the answer. I'll see if I can cook something up. Also, I'm "not really a PHP/WP guy" either 😄

**Comment by pReya** [comment]
@nilsnolde Sorry, but adjusting your plugin seemed like a lot of work. So I decided to create a new plugin on my own, which is a much simpler version of your concept. Hope you don't mind. https://github.com/pReya/wordpress-external-markdown/tree/main#readme

**Comment by nilsnolde** [comment]
:shrug:

**Comment by ElDavoo** [comment]
:(

**Comment by ElDavoo** [comment]
> @nilsnolde Sorry, but adjusting your plugin seemed like a lot of work. So I decided to create a new plugin on my own, which is a much simpler version of your concept. Hope you don't mind. https://github.com/pReya/wordpress-external-markdown/tree/main#readme

thing is, private wiki pages are not supported this way, you still need to auth.

**Comment by pReya** [comment]
@ElDavoo that's correct – it's intended that way. But you can still use `wordpress-markdown-git` for this, right?

**Comment by nilsnolde** [comment]
maybe you can cherry pick @pReya work into a PR here. as he mentioned it's a little more involved, but not too hard if you know php a bit. again, it's fair enough to only implement for now for gitlab, leaving github/bitbucket for others to contribute if they need it
